### PR TITLE
Add whisper

### PR DESCRIPTION
**Pull request by MatthiasReumann**
## Description 
Issue: #13 

## Tasks

- [X] 1) Since the vosk-transcriber is already implemented, the transcriber should be configurable.
- [X] 2) When specified, whisper language models should be downloaded and saved automatically at the beginning of execution.
- [X] 3) Whisper can detect languages automatically. The current request has an attribute called `language`. Find a solution and figure out how to deal with silences at the beginning of the video.
- [x] 4) Make sure to update the `README.md`.

1) Configurable in the config file: `transcriber: [vosk | whisper]`. This allows us to deploy two voice-services using different transcribers (_vosk1.voice.in.tum.de_, _whisper1.voice.in.tum.de_). Then we can use vosk for fast subtitles (e.g. immediately after a VOD upload) and whisper for the overnight generation of high-quality subtitles. _(~~This requires an update of the endpoint in TUM-Live~~)_
2) Whisper language models don't have to be downloaded. 
3) Works automatically. Also: Whisper can detect languages even though in the first 30 seconds music is playing. (I have no idea how)
4) Done.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,22 +1,64 @@
...
+torch==1.13.0
+torchaudio==0.13.0
 tqdm==4.64.0
+transformers==4.24.0
+typing_extensions==4.4.0
 urllib3==1.26.12
+virtualenv==20.16.5
 vosk==0.3.44
 websockets==10.3
+whisper @ git+https://github.com/openai/whisper.git@eff383b27b783e280c089475852ba83f20f64998
```
Comment (joschahenningsen) [inline requirements.txt]: Using this ref requires git to be installed in the docker container

**Review by joschahenningsen — CHANGES_REQUESTED** [review]

File: requirements.txt (hunk 1 above)
Comment (joschahenningsen) [reply requirements.txt]: I actually think we might also need build essentials / gcc

Comment (MatthiasReumann) [reply requirements.txt]: I just realized that the requirements.txt is broken. 🧐 
I will look into the needed dependencies and update the Dockerfile accordingly.

**Comment by joschahenningsen** [comment]
I did some more digging and tried whisper out a bit more. It seems to have [a few issues handling silence](https://github.com/openai/whisper/discussions/29) and produces hilarious results when nothing is said like

```
[01:32:49.600 --> 01:32:52.60]  Copyright WDR 2021
[01:33:19.600 --> 01:33:22.60]  Copyright WDR 2021
[01:33:49.600 --> 01:33:52.60]  Copyright WDR 2021
[01:34:19.600 --> 01:34:21.60]  Copyright WDR 2021
[01:34:49.600 --> 01:34:52.60]  Copyright WDR 2021
[01:35:19.600 --> 01:35:22.60]  Copyright WDR 2021
[01:35:49.600 --> 01:35:52.60]  Copyright WDR 2021
[01:36:19.600 --> 01:36:22.60]  Copyright WDR 2021
[01:36:49.600 --> 01:36:51.60]  Copyright WDR 2021
```

or 

```
[01:10:19.600 --> 01:10:21.60]  I hope you liked the video, don't forget to leave a like and subscribe to the channel, thank you very much for your attention and see you in the next video
```

It seems like setting `--no_speech_threshold 0.2` and `--logprob_threshold None` helps a ton. Can this be incorporated here? If that's not enough we'll need to detect speech, cut the parts of the video and run whisper against them separately.

File: requirements.txt (hunk 1 above)
Comment (MatthiasReumann) [reply requirements.txt]: The issue should be resolved now. _Dockerfile_ + _requirements.txt_ is updated and should work.

**Comment by MatthiasReumann** [comment]
> I did some more digging and tried whisper out a bit more. It seems to have [a few issues handling silence](https://github.com/openai/whisper/discussions/29) and produces hilarious results when nothing is said like
> 
> ```
> [01:32:49.600 --> 01:32:52.60]  Copyright WDR 2021
> [01:33:19.600 --> 01:33:22.60]  Copyright WDR 2021
> [01:33:49.600 --> 01:33:52.60]  Copyright WDR 2021
> [01:34:19.600 --> 01:34:21.60]  Copyright WDR 2021
> [01:34:49.600 --> 01:34:52.60]  Copyright WDR 2021
> [01:35:19.600 --> 01:35:22.60]  Copyright WDR 2021
> [01:35:49.600 --> 01:35:52.60]  Copyright WDR 2021
> [01:36:19.600 --> 01:36:22.60]  Copyright WDR 2021
> [01:36:49.600 --> 01:36:51.60]  Copyright WDR 2021
> ```
> 
> or
> 
> ```
> [01:10:19.600 --> 01:10:21.60]  I hope you liked the video, don't forget to leave a like and subscribe to the channel, thank you very much for your attention and see you in the next video
> ```
> 
> It seems like setting `--no_speech_threshold 0.2` and `--logprob_threshold None` helps a ton. Can this be incorporated here? If that's not enough we'll need to detect speech, cut the parts of the video and run whisper against them separately.

Changing `no_speech_threshold` to  0.275 and `logprob_threshold` to None solved the problem for me locally. We might need to tweak this values later on but for now this should be fine.

**Review by joschahenningsen — APPROVED** [review]
Seems to work nicely! :heart:

**Pull request merged**